sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobsCheckpointStore.cs

```diff
@@ -614,14 +614,25 @@ private static void TryReadLegacyCheckpoint(Span<byte> data, out long? offset, o
                     switch (jsonReader.GetString())
                     {
                         case "Offset":
-                            if (!jsonReader.Read() ||
-                                jsonReader.GetString() is not string offsetString ||
-                                !long.TryParse(offsetString, out long offsetValue))
+
+                            if (!jsonReader.Read())
                             {
                                 return;
                             }
 
-                            offset = offsetValue;
+                            var offsetString = jsonReader.GetString();
+                            if (offsetString != null)
+                            {
+                                if (long.TryParse(offsetString, out long offsetValue))
+                                {
+                                    offset = offsetValue;
+                                }
+                                else
+                                {
+                                    return;
+                                }
+                            }
+
                             break;
                         case "SequenceNumber":
```

Comment:
Do we need similar logic for a null SequenceNumber?

Reply:
The SequenceNumber is a non-nullable value AFAIK